src/sequelize.js

```diff
@@ -1279,7 +1280,7 @@ Sequelize.prototype.Validator = Sequelize.Validator = Validator;
 
 Sequelize.Model = Model;
 
-Sequelize.QueryInterface = QueryInterface;
+Sequelize.QueryInterface = AbstractQueryInterface;
```

Comment:
Last minute bug:

The type export doesn't match the export.
The export is still QueryInterface but it's now typed as AbstractQueryInterface

Reply:
@ephys What should it be? Does it make sense to just change the underlying type without changing the export?

Reply:
We can have it exported as a different name than what we use internally, but the type export & the value export must have the same name

I'd go with QueryInterface outside of Sequelize for now to avoid a breaking change, which can be done by changing how index.d.ts exports QueryInterface

On top of that, the types added to query-interface.types.ts should be public too, need to export them from index.d.ts

Reply:
Implementing this in #15441